File: 43173/backend/internal/model/event.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type EventStatus int

const (
	EventStatusDraft     EventStatus = 0
	EventStatusPublished EventStatus = 1
	EventStatusSoldOut   EventStatus = 2
	EventStatusEnded     EventStatus = 3
	EventStatusCancelled EventStatus = 4
)

type Event struct {
	ID           uint           `json:"id" gorm:"primaryKey"`
	UserID       uint           `json:"user_id" gorm:"index;not null"`
	ArtistID     uint           `json:"artist_id" gorm:"index;not null"`
	Title        string         `json:"title" gorm:"size:200;not null"`
	Description  string         `json:"description" gorm:"type:text"`
	CoverURL     string         `json:"cover_url" gorm:"size:255"`
	Venue        string         `json:"venue" gorm:"size:200;not null"`
	Address      string         `json:"address" gorm:"size:255"`
	City         string         `json:"city" gorm:"size:100"`
	Longitude    float64        `json:"longitude"`
	Latitude     float64        `json:"latitude"`
	StartTime    time.Time      `json:"start_time"`
	EndTime      time.Time      `json:"end_time"`
	DoorTime     *time.Time     `json:"door_time"`
	TicketPrice  float64        `json:"ticket_price"`
	TotalTickets int            `json:"total_tickets"`
	SoldTickets  int            `json:"sold_tickets" gorm:"default:0"`
	MaxPerUser   int            `json:"max_per_user" gorm:"default:4"`
	Status       EventStatus    `json:"status" gorm:"default:0"`
	PublishedAt  *time.Time     `json:"published_at"`
	ViewCount    int64          `json:"view_count" gorm:"default:0"`
	LikeCount    int64          `json:"like_count" gorm:"default:0"`
	ShareCount   int64          `json:"share_count" gorm:"default:0"`
	SeatMap      string         `json:"seat_map" gorm:"type:text"`
	User         *User          `json:"user,omitempty" gorm:"foreignKey:UserID"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `json:"-" gorm:"index"`
}

func (Event) TableName() string {
	return "events"
}

type TicketStatus int

const (
	TicketStatusPending   TicketStatus = 0
	TicketStatusPaid      TicketStatus = 1
	TicketStatusUsed      TicketStatus = 2
	TicketStatusCancelled TicketStatus = 3
	TicketStatusRefunded  TicketStatus = 4
)

type Ticket struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	OrderID     uint           `json:"order_id" gorm:"index;not null"`
	UserID      uint           `json:"user_id" gorm:"index;not null"`
	EventID     uint           `json:"event_id" gorm:"index;not null"`
	SeatRow     int            `json:"seat_row"`
	SeatCol     int            `json:"seat_col"`
	SeatLabel   string         `json:"seat_label" gorm:"size:20"`
	QRCode      string         `json:"qr_code" gorm:"size:255"`
	Status      TicketStatus   `json:"status" gorm:"default:0"`
	UsedAt      *time.Time     `json:"used_at"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

func (Ticket) TableName() string {
	return "tickets"
}

type OrderStatus int

const (
	OrderStatusPending   OrderStatus = 0
	OrderStatusPaid      OrderStatus = 1
	OrderStatusCompleted OrderStatus = 2
	OrderStatusCancelled OrderStatus = 3
	OrderStatusRefunded  OrderStatus = 4
)

type Order struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	OrderNo     string         `json:"order_no" gorm:"uniqueIndex;size:50;not null"`
	UserID      uint           `json:"user_id" gorm:"index;not null"`
	EventID     uint           `json:"event_id" gorm:"index;not null"`
	ArtistID    uint           `json:"artist_id" gorm:"index;not null"`
	TotalAmount float64        `json:"total_amount"`
	Quantity    int            `json:"quantity"`
	Status      OrderStatus    `json:"status" gorm:"default:0"`
	PaymentMethod string       `json:"payment_method" gorm:"size:50"`
	PaidAt      *time.Time     `json:"paid_at"`
	Remark      string         `json:"remark" gorm:"size:255"`
	User        *User          `json:"user,omitempty" gorm:"foreignKey:UserID"`
	Event       *Event         `json:"event,omitempty" gorm:"foreignKey:EventID"`
	Tickets     []Ticket       `json:"tickets,omitempty" gorm:"foreignKey:OrderID"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

func (Order) TableName() string {
	return "orders"
}

File: 43173/frontend/src/api/event.ts
import request, { PageData, PageParams } from '@/utils/request'

export interface Event {
  id: number
  user_id: number
  artist_id: number
  title: string
  description?: string
  cover_url: string
  venue: string
  address?: string
  city?: string
  longitude?: number
  latitude?: number
  start_time: string
  end_time: string
  door_time?: string
  ticket_price: number
  total_tickets: number
  sold_tickets: number
  max_per_user: number
  status: number
  published_at?: string
  view_count: number
  like_count: number
  share_count: number
  seat_map?: string
  user?: {
    id: number
    username: string
    nickname: string
    avatar: string
  }
  created_at: string
  updated_at: string
}

export interface Order {
  id: number
  order_no: string
  user_id: number
  event_id: number
  artist_id: number
  total_amount: number
  quantity: number
  status: number
  payment_method?: string
  paid_at?: string
  remark?: string
  user?: any
  event?: Event
  tickets?: Ticket[]
  created_at: string
  updated_at: string
}

export interface Ticket {
  id: number
  order_id: number
  user_id: number
  event_id: number
  seat_row: number
  seat_col: number
  seat_label: string
  qr_code: string
  status: number
  used_at?: string
  event?: Event
  created_at: string
  updated_at: string
}

export interface CreateEventParams {
  title: string
  description?: string
  cover_url?: string
  venue: string
  address?: string
  city?: string
  longitude?: number
  latitude?: number
  start_time: string
  end_time: string
  door_time?: string
  ticket_price: number
  total_tickets: number
  max_per_user?: number
  seat_map?: string
}

export interface PurchaseTicketParams {
  event_id: number
  quantity: number
  seats?: Array<{ row: number; col: number }>
}

export const eventApi = {
  list: (params: PageParams & { keyword?: string; city?: string; status?: number; artist_id?: number }) => 
    request.get<PageData<Event>>('/events', params),
  getAll: (params: PageParams) => request.get<PageData<Event>>('/events', params),
  getById: (id: number) => request.get<Event>(`/events/${id}`),
  getMyEvents: (params: PageParams) => request.get<PageData<Event>>('/events/my', params),
  getMyTickets: (params: PageParams) => request.get<PageData<any>>('/tickets/my', params),
  create: (data: CreateEventParams) => request.post<Event>('/events', data),
  update: (id: number, data: Partial<CreateEventParams>) => request.put(`/events/${id}`, data),
  delete: (id: number) => request.delete(`/events/${id}`),
  publish: (id: number) => request.post(`/events/${id}/publish`),
  getStats: (id: number) => request.get(`/events/${id}/stats`),
  getSeats: (id: number) => request.get(`/events/${id}/seats`),
  purchase: (data: PurchaseTicketParams) => request.post<Order>('/events/purchase', data),
  getOrders: (params: PageParams) => request.get<PageData<Order>>('/orders', params),
  getOrderById: (id: number) => request.get<Order>(`/orders/${id}`),
  getOrdersByArtist: (artistId: number, params: PageParams) => 
    request.get<PageData<Order>>(`/orders/artist/${artistId}`, params),
  getTickets: (params: PageParams) => request.get<PageData<Ticket>>('/tickets', params),
  useTicket: (id: number) => request.post(`/tickets/${id}/use`)
}
